Modelica/Blocks/Logical.mo

```diff
@@ -358,6 +358,84 @@ Real input u2, otherwise the output is <strong>false</strong>.
 </html>"));
   end LessEqual;
 
+  block IntervalTest
+    "Tests if input u is within the specified interval"
+    extends Modelica.Blocks.Interfaces.partialBooleanSO;
+    parameter Boolean useConstantLimits=true "Use constant limits, if true; otherwise signal input limits are used" annotation(Evaluate=true);
+    parameter Real constantLowerLimit "Lower limit of interval" annotation(Dialog(enable=useConstantLimits));
+    parameter Boolean ClosedOnLeft=false "Include lower limit point, if true" annotation(Evaluate=true);
+    parameter Real constantUpperLimit "Upper limit of interval" annotation(Dialog(enable=useConstantLimits));
+    parameter Boolean ClosedOnRight=false "Include upper limit point, if true" annotation(Evaluate=true);
+    parameter Boolean InsideInterval=true "Output u is equal to true, if u is inside interval; otherwise output is inverted" annotation(Evaluate=true);
+    Modelica.Blocks.Interfaces.RealInput u annotation (Placement(transformation(
+            extent={{-140,-20},{-100,20}}), iconTransformation(extent={{-140,-20},{
+              -100,20}})));
+    Modelica.Blocks.Interfaces.RealInput lowerLimit if not useConstantLimits
+      annotation (Placement(transformation(extent={{-140,-100},{-100,-60}})));
+    Modelica.Blocks.Interfaces.RealInput upperLimit if not useConstantLimits
+      annotation (Placement(transformation(extent={{-140,60},{-100,100}})));
+
+  protected
+    Modelica.Blocks.Interfaces.RealInput lowerLimitInternal annotation (Placement(
+          transformation(extent={{-94,-84},{-86,-76}}), iconTransformation(extent={
+              {-94,56},{-86,64}})));
+    Modelica.Blocks.Interfaces.RealInput upperLimitInternal annotation (Placement(
+          transformation(extent={{-94,76},{-86,84}}), iconTransformation(extent={{-94,76},
+              {-86,84}})));
+  equation
+    if useConstantLimits then
+      upperLimitInternal = constantUpperLimit;
+      lowerLimitInternal = constantLowerLimit;
+    else
+      connect(upperLimitInternal, upperLimit);
+      connect(lowerLimitInternal, lowerLimit);
+    end if;
+    assert(upperLimitInternal>lowerLimitInternal, "Erroneous interval limits");
```

Comment:
If this assertion is really needed, I would set this as a warning only - insted of hard termination of the simulation. Optionally, a parameter of type "AssertionLevel" could be add to give the user an opportunity to decide - but this approach is not used elsewhere in MSL so maybe not that good idea.